internal/kafka/command_replica_list.go

```diff
@@ -1,84 +1,56 @@
+	"github.com/confluentinc/cli/v3/pkg/errors"
 	"github.com/confluentinc/cli/v3/pkg/examples"
 	"github.com/confluentinc/cli/v3/pkg/kafkarest"
 	"github.com/confluentinc/cli/v3/pkg/output"
-	"github.com/confluentinc/cli/v3/pkg/utils"
 )
 
-type replicaHumanOut struct {
-	ClusterId          string `human:"Cluster"`
-	TopicName          string `human:"Topic Name"`
-	BrokerId           int32  `human:"Broker ID"`
-	PartitionId        int32  `human:"Partition ID"`
-	IsLeader           bool   `human:"Leader"`
-	IsObserver         bool   `human:"Observer"`
-	IsIsrEligible      bool   `human:"ISR Eligible"`
-	IsInIsr            bool   `human:"In ISR"`
-	IsCaughtUp         bool   `human:"Caught Up"`
-	LogStartOffset     int64  `human:"Log Start Offset"`
-	LogEndOffset       int64  `human:"Log End Offset"`
-	LastCaughtUpTimeMs string `human:"Last Caught Up Time (ms)"`
-	LastFetchTimeMs    string `human:"Last Fetch Time (ms)"`
-	LinkName           string `human:"Link Name"`
-}
-
-type replicaSerializedOut struct {
-	ClusterId          string `serialized:"cluster_id"`
-	TopicName          string `serialized:"topic_name"`
-	BrokerId           int32  `serialized:"broker_id"`
-	PartitionId        int32  `serialized:"partition_id"`
-	IsLeader           bool   `serialized:"is_leader"`
-	IsObserver         bool   `serialized:"is_observer"`
-	IsIsrEligible      bool   `serialized:"is_isr_eligible"`
-	IsInIsr            bool   `serialized:"is_in_isr"`
-	IsCaughtUp         bool   `serialized:"is_caught_up"`
-	LogStartOffset     int64  `serialized:"log_start_offset"`
-	LogEndOffset       int64  `serialized:"log_end_offset"`
-	LastCaughtUpTimeMs int64  `serialized:"last_caught_up_time_ms"`
-	LastFetchTimeMs    int64  `serialized:"last_fetch_time_ms"`
-	LinkName           string `serialized:"link_name"`
+type ReplicaOut struct {
+	ClusterId   string `human:"Cluster" serialized:"cluster_id"`
+	TopicName   string `human:"Topic Name" serialized:"topic_name"`
+	PartitionId int32  `human:"Partition ID" serialized:"partition_id"`
+	BrokerId    int32  `human:"Broker ID" serialized:"broker_id"`
+	IsLeader    bool   `human:"Leader" serialized:"is_leader"`
+	IsInIsr     bool   `human:"In ISR" serialized:"is_in_isr"`
 }
 
 func (c *replicaCommand) newListCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "list",
-		Short: "List Kafka replica statuses.",
-		Long:  "List partition-replicas statuses filtered by topic and partition.",
-		Args:  cobra.NoArgs,
+		Use:   "list <topic>",
+		Short: "List Kafka replicas.",
+		Long:  "List partition-replicas for a Kafka topic.",
+		Args:  cobra.ExactArgs(1),
 		RunE:  c.list,
 		Example: examples.BuildExampleString(
 			examples.Example{
-				Text: `List the replica statuses for partition 1 of topic "my_topic".`,
-				Code: "confluent kafka replica list --topic my_topic --partition 1",
+				Text: `List the replicas for partition 1 of topic "my-topic".`,
+				Code: "confluent kafka replica list my-topic --partition 1",
```

Comment:
Looking for opinions here.

The current `replica list` uses flags for both `topic` and `partition`. But since `topic` is required, I figured it made more sense to just make it a required argument.

Should I change the new `replica status list` to this style? Or change this one back to two flags?

Reply:
> But since topic is required, I figured it made more sense to just make it a required argument.

This is another thing that's inconsistent across our commands. I think we should only use arguments when the subcommand matches the argument. Here it looks like `my-topic` is the name of a replica or something? I would support adding `--topic` as a required flag here instead.